### PR TITLE
Add Checkout FO to Sanity tests puppeteer

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Adding new Sanity test : Checkout FO
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="sanity/checkoutFO/*" URL_FO=SHOP_URL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/15470)

### CODE REVIEW

**Comment by boubkerbribri** [comment]
do not merge yet please, need #15421 modification on `data/FO/cart.js`

File: tests/puppeteer/campaigns/sanity/checkoutFO/01_orderProduct.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  });
+  test('should go to login page', async () => {
+    await homePage.goToLoginPage();
+    const pageTitle = await loginPage.getPageTitle();
+    await expect(pageTitle).to.equal(loginPage.pageTitle);
+  });
+  test('should sign In in FO With default account', async () => {
+    await loginPage.customerLogin(customer.defaultAccount);
+    const connected = await homePage.isCustomerConnected();
+    await expect(connected, 'Customer is not connected in FO').to.be.true;
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/sanity/checkoutFO/01_orderProduct.js]: Why the sentence "is not connected in FO" should be true ? That means you're not connected, right ?

**Comment by SimonGrn** [comment]
Thanks @boubkerbribri

File: tests/puppeteer/pages/FO/checkout.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
+const CommonPage = require('../commonPage');
+
+module.exports = class Checkout extends CommonPage {
+  constructor(page) {
+    super(page);
+
```
Comment (nesrineabdmouleh) [inline tests/puppeteer/pages/FO/checkout.js:6]: Please delete the empty line

Comment (boubkerbribri) [reply tests/puppeteer/pages/FO/checkout.js:6]: done thanks

File: tests/puppeteer/pages/FO/checkout.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  /**
+   * Choose payment method and validate Order
+   * @param paymentModuleName, payment method chosen (ex : ps_wirepayment)
+   * @return {Promise<void>}
+   */
+  async choosePaymentAndOrder(paymentModuleName) {
+    await this.page.click(this.paymentOptionInput.replace('%NAME', paymentModuleName));
+    await Promise.all([
+      this.page.waitForSelector(this.paymentConfirmationButton, {visible: true}),
```
Comment (nesrineabdmouleh) [inline tests/puppeteer/pages/FO/checkout.js]: You can use the function `waitForSelectorAndClick` from commonPage

File: tests/puppeteer/campaigns/data/FO/customer.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+
```
Comment (Progi1984) [inline tests/puppeteer/campaigns/data/FO/customer.js]: Remove empty line

File: tests/puppeteer/pages/FO/home.js [hunk 5]
```diff
@@ -56,4 +66,65 @@ module.exports = class Home extends CommonPage {
...
+      this.page.hover(this.productImg.replace('%NUMBER', id)),
+    ]);
+    await Promise.all([
+      this.page.waitForSelector(this.quickViewModalDiv),
+      this.page.$eval(this.productQuickViewLink.replace('%NUMBER', id), el => el.click()),
+    ]);
+  }
+
+  /**
+   * Add
```
Comment (Progi1984) [inline tests/puppeteer/pages/FO/home.js]: Add what ?

File: tests/puppeteer/campaigns/data/FO/customer.js (hunk 4 above)
Comment (boubkerbribri) [reply tests/puppeteer/campaigns/data/FO/customer.js]: Done thanks

File: tests/puppeteer/pages/FO/home.js (hunk 5 above)
Comment (boubkerbribri) [reply tests/puppeteer/pages/FO/home.js]: Comment completed, thanks

**Review by Progi1984 — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thanks @boubkerbribri

**Pull request merged**